src/Event/DatasetIndexSubscriber.php

```diff
@@ -119,6 +120,12 @@ public function populateCalculatedFields(TransformEvent $event)
                 $document->set('collectionEndDate', $collectionEndDate->format('Y-m-d H:i:s'));
             }
         }
+
+        // For DIF only, set DOI to a DOI with empty string as the doi value.
+        if (false === $dataset->hasDatasetSubmission()) {
```

Comment:
We usually use instanceOf to check if its an object and if its not null, and we also need to check if the dataset has been submitted

Reply:
hasDatasetSubmission returns a bool, so used it over $dataset->getDatasetSubmission(). I forgot about the unsubmitted...will fix.